lib/oci_utils/kvm/virt.py

```diff
@@ -142,8 +142,8 @@ def find_unassigned_vf_by_phys(phys, domain_interfaces, desired_mac):
     """
     configured = sysconfig.read_network_config()
     ifaces = nic.get_interfaces()
-    virt_fns = ifaces[phys].get('virtfns', {})
-    vfs = {virt_fns[v]['mac']: (virt_fns[v]['pciId'], v) for v in virt_fns}
+    virt_fns = ifaces[phys].get('virt_fns', {})
```

Comment:
This change seem very unrelated, was there a bug here?

Reply:
This was a regression I've found.  key are indeed 'virt_fns' . 
the discovery of free resource was broken